pkg/testrunner/types.go

```diff
@@ -122,11 +122,26 @@ type TestrunSummary struct {
 
 // StepSummary is the result of a specific step.
 type StepSummary struct {
-	Metadata  *Metadata        `json:"tm,omitempty"`
-	Type      SummaryType      `json:"type,omitempty"`
-	Name      string           `json:"name,omitempty"`
-	StepName  string           `json:"stepName,omitempty"`
-	Phase     argov1.NodePhase `json:"phase,omitempty"`
-	StartTime *metav1.Time     `json:"startTime,omitempty"`
-	Duration  int64            `json:"duration,omitempty"`
+	Metadata    *Metadata        `json:"tm,omitempty"`
+	Type        SummaryType      `json:"type,omitempty"`
+	Name        string           `json:"name,omitempty"`
+	StepName    string           `json:"stepName,omitempty"`
+	Phase       argov1.NodePhase `json:"phase,omitempty"`
+	StartTime   *metav1.Time     `json:"startTime,omitempty"`
+	Duration    int64            `json:"duration,omitempty"`
+	PreComputed *StepPreComputed `json:"pre,omitempty"`
+}
+
+// StepPreComputed contains fields that could be created at runtime via scripted fields, but are created statically for better performance and better support of grafana
+type StepPreComputed struct {
+	// same as StepSummary.Phase but mapping states to ints (Failed&Timeout -> 0, Succeeded -> 100); allows to do averages on success rate in dashboards
+	PhaseNum int `json:"phaseNum,omitempty"`
```

Comment:
We should change this to a pointer because otherwise all testruns that are `phase != sucess` will be 0 as the struct will be automatically initialized with a value of 0.
If we change the type to a pointer there will be a difference between 0 and nil.

```suggestion
	PhaseNum *int `json:"phaseNum,omitempty"`
```

Reply:
Good catch! Thanks!